### ISSUE TITLE
Optional auto scrolling

### ISSUE THREAD

**Issue by vladimir-konnov** [atlassian/react-beautiful-dnd#460]
## Bug or feature request?
Feature

### Expected behavior
There should be an option to disable auto scrolling since it's not required in all cases
### Actual behavior
When there's another control (not droppable) below the kanban board there's no sense in scrolling the page down. In my case it's a disaster, because on scrolling droppable coordinates/dimensions aren't recalculated so dragover (and drop too) doesn't work properly
https://monosnap.com/file/n72kQ8y0evWzzSmAgAXdwJZsCsYsuS
### Demo
https://monosnap.com/file/n72kQ8y0evWzzSmAgAXdwJZsCsYsuS

**Comment by alexreardon** [comment]
You could possibly opt out yourself by adding your own capture, non-passive scroll listener to the window and calling `event.preventDefault()`. This would prevent all scrolling during a drag

**Comment by ssmlee04** [comment]
@alexreardon I might be having other components that does drag & drop and has it's own logic and it wouldn't make sense to disable scrolling for other components at the same time.

**Comment by matttk** [comment]
Would also like to be able to disable autoscroll. I'm searching through all the auto scroll bugs to find out what's going wrong for me but haven't come up with anything yet. Seems like a good temporary solution to just kill autoscrolling, especially since I don't need it.

I have a horizontal list, basically made from the basic horizontal list example. However, for some reason, as soon as I pick up an item, the page starts scrolling to the top. The faster I move the item, the faster it scrolls to the top. I'm nowhere near the top...

**Comment by uffou** [comment]
I also need to be able to disable auto scroll. I have a horizontal scrolling container and another one that I am dragging items from. At the moment when I start dragging and intersect the horizontal scrolling container, it starts to scroll which is not the desired behavior. A simple `autoScroll={false}` would do a great deal for everyone.

**Comment by buzzpsych** [comment]
Where in the codebase are you powering the auto scroll? Can this be made optional? 

I currently have a fixed sidebar for example with some elements in it that are recordable. Whenever I try and move down or up it moves my body.

The best solution I found so far is adding addEventListener for scroll with 
`function noscroll() {
  window.scrollTo(0, 0)
}` 

BUT the problem is that if you are already scrolled down the page and you initiate this, it jumps back up. BAD UX

**Comment by cocacrave** [comment]
So far I tried `onDragStart` and `onDragEnd` calls the function to toggle classname on body to apply `overflow: hidden` if the `item.source.droppableId` starts with a string `sidebar` and it's toggling the class fine but why is body still scrolling? My sidebar is a sticky component, and when I try to drag an item towards the top or bottom, the whole body scrolls.

Any workaround?

**Comment by Venryx** [comment]
One way to accomplish this is with the string-replace-webpack-plugin: https://github.com/jamesandersen/string-replace-webpack-plugin

My webpack config:
```
webpackConfig.module.rules.push({
	// test: /connected-(draggable|droppable).js$/,
	test: /react-beautiful-dnd.esm.js$/,
	loader: StringReplacePlugin.replace({ replacements: [
		// disable map edge-scrolling, when option is set
		{
			// pattern: /var canScrollDroppable = function canScrollDroppable\(droppable, change\) {/,
			pattern: /var canScrollDroppable = function canScrollDroppable.+/,
			replacement: () => `
				var canScrollDroppable = function canScrollDroppable(droppable, change) {
					if (window.LockMapEdgeScrolling()) return false;
			`.trim(),
		},
	] }),
});
```

Of course, you have to also implement the LockMapEdgeScrolling global function.

**Comment by cavaloni** [comment]
This is a desired feature for us as well. Adding comment for a "vote", and a strong one at that.

**Comment by cavaloni** [comment]
Figured out a solution for those where `scrollTo(0, 0)` is no good because its lower in the page, which is simply to get the `window.scrollY` position in `onBeforeDragStart` and setting event listener there, as noted above in @buzzpsych response, and just simply setting the `y` argument in `scrollTo` to the current scroll position. Then it stays where you are! As it should, if we could just have a damn option to turn off this feature.

**Comment by mjuopperi** [comment]
I fiddled with this a bit and it was fairly easy to add a prop to `DragDropContext` to disable auto scrolling. Contribution docs mention that they prefer people not to open PR:s directly, so not opening one yet. 

Here is how I did it: https://github.com/mjuopperi/react-beautiful-dnd/commits/optional-autoscroll

**Comment by olee** [comment]
@alexreardon would you mind reopening this issue?
I think a lot of users - myself included - would just like to prevent the autoScroll from triggering in certain circumstances.

In my case it is a table header which is automatically scrolled together with the table body via an event handler which is why it does not have any overflow. This now makes the drag handler pick some random parent container for auto scrolling which is not desired at all.

A simple option to disable the lookup for a scrollable container about [here](https://github.com/atlassian/react-beautiful-dnd/blob/master/src/view/use-droppable-publisher/use-droppable-publisher.js#L131) would be sufficient.

Even better however would by imho to allow users to manually provide a scroll container. This would even allow use cases such as mine where the scroll container actually does not have any overflow attribute because it is scrolled by code.

**Comment by zachhale** [comment]
I also have a situation where I want to only allow the person to drag between the items in the current view and not allow them to scroll between the items hidden via `overflow-x: hidden;`.

**Comment by cryy** [comment]
this issue should be reopened, it's crazy how the library still doesn't have a way to disable autoscroll almost 3 years after this issue. editing the library internals is a hacky way.

**Comment by AaronLayton** [comment]
Does anyone know if this issue got reopened else where? There looks to be strong community support for being able to disable scrolling.

I have come across a bug when the page scrolls one of the Droppable containers stops responding to drag over events

**Comment by KatFishSnake** [comment]
Any updates on this issue?

**Comment by olee** [comment]
@alexreardon would be great if you can at least take the time to just reopen this issue considering the feedback received here.

**Comment by K4n3dam4** [comment]
Agreed. I'd also like to see this issue reopened. I can think of a bunch of situations where the option to disable auto scroll could be needed. I'm in one of these atm and would prefer not having to use one of the workarounds mentioned above.

**Comment by rawcomposition** [comment]
I also need a way to turn this behavior off. I have a scrollable sidebar and autoscroll sometimes incorrectly scrolls the body element, resulting in a completely messed up interface. For me, this issue is also related to this unresolved problem: https://github.com/clauderic/dnd-kit/issues/298

**Comment by mnpenner** [comment]
Mine is just outright breaking. I'm getting

```
Uncaught TypeError: Cannot read properties of undefined (reading 'stop')
    at react-beautiful-dnd.esm.js:3832:24
```

Which is here:

```
var autoScroll = (function (autoScroller) {
  return function (store) {
    return function (next) {
      return function (action) {
        if (shouldStop(action)) {
          autoScroller.stop();  // <--- autoScroller is undefined
          next(action);
          return;
        }

        if (action.type === 'INITIAL_PUBLISH') {
          next(action);
          var state = store.getState();
          !(state.phase === 'DRAGGING') ? process.env.NODE_ENV !== "production" ? invariant(false, 'Expected phase to be DRAGGING after INITIAL_PUBLISH') : invariant(false) : void 0;
          autoScroller.start(state);
          return;
        }

        next(action);
        autoScroller.scroll(store.getState());
      };
    };
  };
});
```

Only happens in prod which makes this pretty hard to debug. Still working on it. I don't think I need autoScroll so I'd love to just disable it and make this problem go away.

**Comment by KiyeopYang** [comment]
why this is not solved until now?
It throw error when I use nested vertical list with virtualized.

**Comment by kangwuyi** [comment]
me too...
